### PR TITLE
GUVNOR-2602: Guided Rule Editor: Facts created in RHS used in RHS bindings are not retained on re-opening rule

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/GUVNOR-2602

### CODE REVIEW

File: drools-workbench-models/drools-workbench-models-commons/src/main/java/org/drools/workbench/models/commons/backend/rule/RuleModelPersistenceHelper.java [hunk 1]
```diff
@@ -319,7 +319,8 @@ static String inferDataType( final ActionFieldList action,
         }
         for ( ModelField modelField : modelFields ) {
             if ( modelField.getName().equals( field ) ) {
-                return modelField.getType();
+                return getSimpleFactType( modelField.getType(),
```
Comment (manstis) [inline drools-workbench-models/drools-workbench-models-commons/src/main/java/org/drools/workbench/models/commons/backend/rule/RuleModelPersistenceHelper.java:322]: This is the fix.

File: drools-workbench-models/drools-workbench-models-commons/src/test/java/org/drools/workbench/models/commons/backend/rule/RuleModelDRLPersistenceUnmarshallingTest.java [hunk 2]
```diff
@@ -8534,4 +8535,83 @@ public void compositeFieldConstraintWithOnePredicateAndOneLiteral() {
                       fp_cfp_sfp2.getConstraintValueType() );
     }
 
+    @Test
```
Comment (manstis) [inline drools-workbench-models/drools-workbench-models-commons/src/test/java/org/drools/workbench/models/commons/backend/rule/RuleModelDRLPersistenceUnmarshallingTest.java:8538]: This is the test.

**Comment by manstis** [comment]
I've highlighted the fix and the test.. the rest is just tidying up existing tests (as IDEA lit up like a 1KW lightbulb with warnings in the test class).

**Comment by mrietveld** [comment]
+1 when the tests complete.

**Pull request merged**